### PR TITLE
feat(hotspot-text-style-tab): added text style tab

### LINKED ISSUE #1768: [HotspotTextStyleTab] New subcomponent for the HotSpotEditorModal text style tab
Part of issue #1753

### What package is this for?

- [x] React
- [ ] Angular

### Summary
A sub component needed for #1753 to configure the label texts

**Additional context** 
image

_Carbon add-ons for Watson IoT is a collaborative system. We encourage teams to
build components and submit them for integration._

### PR DESCRIPTION
**Pull request by marcJV**
Closes #1768

**Summary**

- Added Hotspot Text Style Tab

**Acceptance Test (how to verify the PR)**

- Go to `?path=/story/watson-iot-experimental-hotspoteditormodal-hotspottextstyletab--default`
Verify design.
Verify updates update in the actions tab.

### CODE REVIEW

File: src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+          }}
+        />
+      </div>
+
+      <div className={`${iotPrefix}--hotspot-text-style-tab__row`}>
+        <ColorDropdown
+          id={`${iotPrefix}--hotspot-text-style-tab__background-color`}
+          titleText={backgroundLabel}
+          light={light}
+          selectedColor={background.color ?? backgroundColors?.[0]}
```
Comment (bjornalm) [inline src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx:195]: Suggestion, skip the fallback `backgroundColors?.[0]` and just use `backgroundColor` which might be undefined which is okay.

File: src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+          min={0}
+          max={100}
+          value={background.opacity}
+          ref={fillOpacityRef}
+          step={1}
+          light={light}
+          invalidText={fillOpacityInvalid}
+          onChange={() => {
+            onChange({
+              background: { opacity: fillOpacityRef.current.value },
```
Comment (bjornalm) [inline src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx:216]: It seemed weird that we would have to use a ref to obtain the value, so I looked at the emitted event and it appears that carbon has added an `imaginaryTarget` element prop that contains the value. So you could do something like

```
onChange={(event) => {
  onChange({ background: event.imaginaryTarget.value})
}}
```

File: src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    fontSizeInvalid,
+    backgroundLabel,
+    fillOpacityLabel,
+    fillOpacityInvalid,
+    borderLabel,
+    borderWidthLabel,
+    borderWidthInvalid,
+  } = merge({}, defaultProps.i18n, i18n);
+
+  const fontSizeRef = useRef(null);
```
Comment (bjornalm) [inline src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx:118]: We can probably remove these refs, se comment below.

**Review by bjornalm — CHANGES_REQUESTED** [review]
Nice work! I have some suggestions though

File: src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    backgroundLabel: PropTypes.string,
+    fillOpacityLabel: PropTypes.string,
+    fillOpacityInvalid: PropTypes.string,
+    borderLabel: PropTypes.string,
+    borderWidth: PropTypes.string,
+    borderWidthInvalid: PropTypes.string,
+  }),
+  /** Callback for when any of the form element's value changes */
+  onChange: PropTypes.func.isRequired,
+  /** The state values of the controlled form elements e.g. { title: 'My hotspot 1', description: 'Lorem ipsum' } */
```
[the remark this answers is not in the document]
Comment (bjornalm) [reply src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx]: There is no 'title' prop for the "formValues" object of this component...

**Review by bjornalm — CHANGES_REQUESTED** [review]
It looks like several of the previous issues were marked as resolved without being fixed or commented on.

**Comment by bjornalm** [comment]
Look good. Just one thing I noticed the there are two RTL issues with the content switcher.


image


I've solved the text padding issue in this [pr](https://github.com/carbon-design-system/carbon-addons-iot-react/pull/1836) which also contained a mixin for the border radius. Perhaps we should move that mixin out so that it can be reused between components. Have a look here https://github.com/carbon-design-system/carbon-addons-iot-react/blob/next/src/components/ImageGalleryModal/_image-gallery-modal.scss#L3-L21

**Review by tay1orjones — APPROVED** [review]
Looks good once @bjornalm's feedback is addressed

**Comment by davidicus** [comment]
@marcJV can we resolve these conflicts so this can be merged when @bjornalm reviews?

**Review by bjornalm — APPROVED** [review]

**Pull request merged**